misaligned.cpp

```diff
@@ -1,21 +1,65 @@
+//#include "PrintColorMapTest.h"
+//#include "PrintColorMapTest.h"
+//#include "ColorMap.h"
 #include <iostream>
+#include <vector>
 #include <assert.h>
+#include<string>
+#include <iomanip>
+struct ColorPair
+    {
+        int number;
+        std::string majorColor;
+        std::string minorColor;
+    };
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+std::vector<ColorPair> PrintColorMap() {
+    const char* majorColor[] = { "White", "Red", "Black", "Yellow", "Violet" };
+    const char* minorColor[] = { "Blue", "Orange", "Green", "Brown", "Slate" };
     int i = 0, j = 0;
-    for(i = 0; i < 5; i++) {
-        for(j = 0; j < 5; j++) {
-            std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
+    std::vector<ColorPair> colorMap;
+        for (i = 0; i < 5; i++)
+    {
+        for (j = 0; j <5; j++)
+        {
+            ColorPair cp;
+            cp.number = i * 5 + j+1;
+            cp.majorColor = majorColor[i];
+            cp.minorColor = minorColor[j];
+            colorMap.push_back(cp);
+            std::cout << std::setw(2)<<cp.number << std::setw(2)<< "|" << std::setw(6)<<cp.majorColor << std::setw(2)<< "|" << std::setw(6) << cp.minorColor << "\n";
         }
     }
-    return i * j;
+    return colorMap;
 }
 
-int main() {
-    int result = printColorMap();
+void test_PrintColorMap()
+{
+     std::vector<ColorPair> colorMap = PrintColorMap();
+    int result = colorMap.size();
+    std::string expectedstr = "25|Violet|Slate";
+    std::string actualstr = std::to_string(colorMap[24].number) + "|" + colorMap[24].majorColor + "|" + colorMap[24].minorColor;

```

Comment:
This is actually forming the line to be printed. It's part of the deliverable, not the test code.
Test code is meant to prove the functionality, not be delivered to a customer.